### PR TITLE
KPL DaemonException --> (╯°□°)╯︵ ┻━┻

### PR DESCRIPTION
**Pull request by cb372**
If we hit a DaemonException:

1. Log an error
2. Terminate the app
3. Terminate the EC2 instance
4. Delete the GitHub repo
5. Quit the company
6. Flee the country

### CODE REVIEW

**Comment by LATaylor-guardian** [comment]
I feel dirty 👍

File: app/com/gu/eventsforwarder/event/EventForwarder.scala [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
+        logger.error("!!! THE SKY IS FALLING !!! Killing the whole app because we encountered this exception. It's probably issue awslabs/amazon-kinesis-producer#39", e)
+        // We can't recover from this, so we kill the app, mark our own AWS instance as unhealthy and let the ASG replace it
+        try {
+          val request = new SetInstanceHealthRequest()
+            .withHealthStatus("Unhealthy")
+            .withInstanceId(EC2MetadataUtils.getInstanceId)
+            .withShouldRespectGracePeriod(false)
+          autoScalingClient.setInstanceHealth(request)
+        } finally {
+          sys.exit(1)
```
Comment (sihil) [inline app/com/gu/eventsforwarder/event/EventForwarder.scala]: Surely all you need to do here is fail the healthcheck and rely on ASG magic to do the rest?

Comment (cb372) [inline app/com/gu/eventsforwarder/event/EventForwarder.scala]: It turns out we're actually using an EC2 healthcheck for this app, not an ELB one. So our healthcheck endpoint is not even being used for autoscaling. I'll fix that, as I'm pretty sure it's a typo. But I'll also leave this code as it is so we can skip the grace period and replace the machine slightly more quickly.

Comment (LATaylor-guardian) [inline app/com/gu/eventsforwarder/event/EventForwarder.scala]: Good spot.

**Pull request merged**